torch/fx/passes/split_module.py

```diff
@@ -31,6 +31,67 @@ def split_module(
     root_m: torch.nn.Module,
     split_callback: Callable[[torch.fx.node.Node], int],
 ):
+    """
+    Creates subgraphs out of main graph
+
+    Args:
+        m (GraphModule): Graph module to split
+        root_m (torch.nn.Module): root nn module. Not currently used. Included
+            because the the root nn module is usually transformed via
+            torch.fx._symbolic_trace.symbolic_trace, see example below
+        split_callback (Callable[[torch.fx.node.Node], int]): Callable function
+            that maps a given Node instance to a numeric partition identifier.
+            split_module will use this function as the policy for which operations
+            appear in which partitions in the output Module.
+
+    Returns:
+        GraphModule: the module after split.
+
+    Example:
+
+        This is a sample setup:
+
+            class MyModule(torch.nn.Module):
+                def __init__(self):
+                    super().__init__()
+                    self.linear = torch.nn.Linear(4, 4)
+
+                def forward(self, a, b, c, d):
+                    add_1 = a + b
+                    add_2 = add_1 + c
+                    linear_1 = self.linear(add_1)
+                    add_3 = add_2 + d
+                    add_4 = add_2 + linear_1
+                    add_5 = add_3 + add_4
+                    return add_5
+
+            a, b, c, d = torch.ones(4), torch.ones(4), torch.ones(4), torch.ones(4)
+            mm = MyModule()
+            traced = symbolic_trace(mm)
+
+            def split_cb(node: torch.fx.Node) -> int:
+                if node.name == "a" or node.name == "b" or node.name == "add":
```

Comment:
Node names aren't stable in FX so it's not a good idea to match on them. I suggest changing this to something like:

```
if node.op == 'placeholder' and node.target in {'a', 'b'}:
  return 0
elif node.target is torch.add:
  return 0
else:
  return 1
```